core/src/main/java/com/linecorp/armeria/common/metric/RollingHdrQuantiles.java

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ *
+ *  Copyright 2016 Vladimir Bukhtoyarov
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.linecorp.armeria.common.metric;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.LongSupplier;
+import java.util.function.Supplier;
+
+import org.HdrHistogram.DoubleHistogram;
+import org.HdrHistogram.DoubleRecorder;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+
+import io.micrometer.core.instrument.stats.quantile.Quantiles;
+
+final class RollingHdrQuantiles implements Quantiles {
+
+    private static final double[] PREDEFINED_QUANTILES = { 0, 0.5, 0.75, 0.9, 0.95, 0.98, 0.99, 0.999, 1.0 };
+    private static final Collection<Double> MONITORED;
+
+    static {
+        final ImmutableList.Builder<Double> builder = ImmutableList.builder();
```

Comment:
nit. Think we can put list directly than making builder.

```java
MONITORED = Arrays.stream(PREDEFINED_QUANTILES).boxed()
                  .collect(toImmutableList());
```

Reply:
Great idea. Fixed!